### DISCUSSION TITLE
Issue when I want to use RoslynCodeEditor with Avalonia

### DISCUSSION THREAD

**Discussion by Inerska** [AvaloniaUI/Avalonia#6222] [Q&A] [answered]
Hello when I want to use RoslynPad (CodeEditor) in my Avalonia project, it crashes and gives me exception `System.MissingMethodException : 'Method not found: 'System.Action Avalonia.Controls.Primitives.ILogicalScrollable.get_InvalidateScroll()'.'
`

```csproj
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
    <Nullable>enable</Nullable>
    <AutoGenerateBindingRedirects>True</AutoGenerateBindingRedirects>
  </PropertyGroup>
  <ItemGroup>
    <Folder Include="Models\" />
    <AvaloniaResource Include="Assets\**" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Avalonia" Version="0.10.6" />
    <PackageReference Include="Avalonia.Desktop" Version="0.10.6" />
    <PackageReference Include="Avalonia.Diagnostics" Version="0.10.6" />
    <PackageReference Include="Avalonia.ReactiveUI" Version="0.10.6" />
    <PackageReference Include="ReactiveUI.Fody" Version="14.1.1" />
    <PackageReference Include="RoslynPad.Editor.Avalonia" Version="1.2.0" />
    <PackageReference Include="RoslynPad.Roslyn.Avalonia" Version="3.6.0" />
  </ItemGroup>
</Project>
```
`xmlns:editor="clr-namespace:RoslynPad.Editor;assembly=RoslynPad.Editor.Avalonia"`

`<editor:RoslynCodeEditor />`

Thanks in advance!

**Comment by maxkatz6** [comment, marked as answer]
It seems this control was not updated to work with latest Avalonia.

**Reply by maxkatz6** [reply]
Yes, it works only with 0.9.11, and not with 0.10.*
https://www.nuget.org/packages/RoslynPad.Roslyn.Avalonia

**Reply by Inerska** [reply]
So do I have to downgrade all Avalonia packages to 0.9.11 ?

**Reply by maxkatz6** [reply]
You can, but I wouldn't recommend using old Avalonia versions unless you really have to. 

Somebody needs to update this package to newer version. Either its maintainer or somebody from community with a PR.

**Reply by Inerska** [reply]
Can you give me what should I have there ?
```xml
  <ItemGroup>
    <PackageReference Include="Avalonia" Version="0.10.6" />
    <PackageReference Include="Avalonia.Desktop" Version="0.10.6" />
    <PackageReference Include="Avalonia.Diagnostics" Version="0.10.6" />
    <PackageReference Include="Avalonia.ReactiveUI" Version="0.10.6" />
    <PackageReference Include="ReactiveUI.Fody" Version="14.1.1" />
    <PackageReference Include="RoslynPad.Editor.Avalonia" Version="1.2.0" />
    <PackageReference Include="RoslynPad.Roslyn.Avalonia" Version="3.6.0" />
  </ItemGroup>
```
I want to downgrade version in waiting the RoslynPad maintener to update

**Reply by maxkatz6** [reply]
Avalonia 0.9.13 version is latest that should work
Diagnostics package should be removed. There also were breaking changes, see Wiki on this repo. And you won't be able to use new features from 0.10.*.

**Reply by Inerska** [reply]
Oh I see thanks, do you know some code editor for Avalonia ? Rather than AvaloniaEdit ?
I wanted to use TextBox, but it is to limited, and there are some issues like the tab character

**Reply by maxkatz6** [reply]
Not really.
You can try to find something here https://github.com/AvaloniaCommunity/awesome-avalonia
